packages/components/src/internal/components/domainproperties/actions.ts

```diff
@@ -435,31 +435,47 @@ export function addDomainField(domain: DomainDesign, fieldConfig: Partial<IDomai
     }) as DomainDesign;
 }
 
-function updateErrorIndexes(removedFieldIndex: number, domainException: DomainException) {
-    const errorsWithNewIndex = domainException.errors.map(error => {
+function updateErrorIndexes(removedFieldIndexes: Number[], domainException: DomainException) {
+    // Given sequence of field indexes that have existing errors, 'rowIndex', and
+    // array of indexes of removed fields, 'removedFieldIndexes' we must update each
+    // rowIndex based on if its position has been affected by a deleted field.
+    // Eg, rowIndex set [1, 4, 7, 8] with removedFieldIndexes [2, 5] becomes [1, 3, 5, 6].
+    const errorsWithNewIndexes = domainException.errors.map(error => {
         const newRowIndexes = error.rowIndexes.map(rowIndex => {
-            if (rowIndex > removedFieldIndex) {
-                return rowIndex - 1;
+            for (let i = 0; i < removedFieldIndexes.length; i++) {
+                if (i !== removedFieldIndexes.length && removedFieldIndexes[i+1] < rowIndex) {
+                    continue;
+                } else if (rowIndex > removedFieldIndexes[i]) {
+                    return rowIndex-(i+1);
+                } else {
+                    return rowIndex;
+                }
             }
         });
         return error.set('rowIndexes', newRowIndexes);
     });
-
-    return domainException.set('errors', errorsWithNewIndex);
+    return domainException.set('errors', errorsWithNewIndexes);
 }
 
-export function removeField(domain: DomainDesign, index: number): DomainDesign {
-    domain = updateDomainException(domain, index, undefined);
+export function removeFields(domain: DomainDesign, deletableSelectedFields: Number[]) {
```

Comment:
was the `DomainDesign` return type intentionally removed?

Reply:
No—re-added